### PR TITLE
Update new docker jobs to make ECR optional

### PR DESCRIPTION
**Pull request by abhinavGupta16**
### Description
Currently, the new docker build and release jobs have ECR release mandatory. This PR makes that optional with a boolean param
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — COMMENTED** [review]
Not too much questions, just wondering can we have a switch to only do ecr as well?

Right now it is either hub, or hub+ecr.

It would be better to run either hub, or ecr, or both.

Thanks.

**Comment by abhinavGupta16** [comment]
> Not too many questions, just wondering can we have a switch to only do ecr as well?
> 
> Right now it is either hub, or hub+ecr.
> 
> It would be better to run either hub, or ecr, or both.
> 
> Thanks.

@peterzhuamazon - I can add a dropdown list for it. WDYT?

**Comment by abhinavGupta16** [comment]
Added a dropdown after offline discussion

File: tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-dockerhub-only.jenkinsfile.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+   docker-build-with-ecr.run()
+      docker-build-with-ecr.legacySCM(groovy.lang.Closure)
+      docker-build-with-ecr.library({identifier=jenkins@20211123, retriever=null})
+      docker-build-with-ecr.pipeline(groovy.lang.Closure)
+         docker-build-with-ecr.timeout({time=3, unit=HOURS})
```
Comment (gaiksaya) [inline tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-dockerhub-only.jenkinsfile.txt:5]: Needs to be updated to 1 hour

File: tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-only-ecr.jenkinsfile.txt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+   docker-build-with-ecr.run()
+      docker-build-with-ecr.legacySCM(groovy.lang.Closure)
+      docker-build-with-ecr.library({identifier=jenkins@20211123, retriever=null})
+      docker-build-with-ecr.pipeline(groovy.lang.Closure)
+         docker-build-with-ecr.timeout({time=3, unit=HOURS})
```
Comment (gaiksaya) [inline tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-only-ecr.jenkinsfile.txt:5]: Same here

File: tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-dockerhub-only.jenkinsfile.txt (hunk 1 above)
Comment (abhinavGupta16) [reply tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-dockerhub-only.jenkinsfile.txt]: Good spot. I missed committing these files.

File: tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-only-ecr.jenkinsfile.txt (hunk 2 above)
Comment (abhinavGupta16) [reply tests/jenkins/jenkinsjob-regression-files/docker-ecr/docker-build-with-only-ecr.jenkinsfile.txt]: Good spot. I missed committing these files.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**